### PR TITLE
fix(errors): Make user column translation a processor

### PR DESCRIPTION
**Pull request by lynnagara**
Make user column translation a query processor. This ensures that the
proper translation is applied to both queries to the "user" column as
well as those that reference the "tags[sentry:user]" promoted tag column.

### CODE REVIEW

File: snuba/datasets/storages/user_column_processor.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+from snuba.query.expressions import Column, Expression, FunctionCall, Literal
+from snuba.request.request_settings import RequestSettings
+
+
+class UserColumnProcessor(QueryProcessor):
+    """
+    Return null instead of empty user to align errors to events storage behavior.
+    This translation is applied as a column processor rather than a translator, so
+    that it will be properly applied to the promoted sentry:user tag.
+    """
```
Comment (fpacifici) [inline snuba/datasets/storages/user_column_processor.py:12]: I have a few questions around this behavior.
If we are concerned about promoting the `sentry:user` tag why are we exposing the user column at all? Shouldn't we either resolve `user` in the logical schema in to `tags[sentry:user]` ? (In the translators you modified, the user column was expose as a column).

Comment (lynnagara) [reply snuba/datasets/storages/user_column_processor.py:12]: Both "user" and "tags[sentry:user]" appear to be used by the Sentry app currently - this solution ensures the same behaviour whichever is used. "User" should still be exposed as a column.

Comment (fpacifici) [reply snuba/datasets/storages/user_column_processor.py:12]: Ok, but then the user column should resolve to the tag not directly to the column.

Comment (lynnagara) [reply snuba/datasets/storages/user_column_processor.py:12]: Added the column mapping https://github.com/getsentry/snuba/pull/1593/commits/1df3ab8e9775c19df71f19c7b3c0d9d18ae5bbce - was that what you meant? The "user" column now gets mapped to the tag first then gets mapped back to the column.

Comment (fpacifici) [reply snuba/datasets/storages/user_column_processor.py:12]: Yes, thanks. Now this means that there is no path in the query procesing where we need to deal with both the user column and the user tag.

**Review by fpacifici — APPROVED** [review]

**Pull request merged**